protocol/avdl/stellar1/local.avdl

```diff
@@ -335,6 +335,59 @@ protocol local {
     boolean pending;
   }
 
+  // Check but do not send a request.
+  // Arguments are optional unless marked [required] or otherwise described.
+  // However, a coherent payment is required in order for res.readyToSend to returned as true.
+  BuildRequestResLocal buildRequestLocal(
+    int sessionID,
+
+    // Recipient username, acctID, or assertion.
+    // Examples: "", "russel", "GBQTE2V7Y356TFBZL6YZ2PA3KIILNSAAQRV5C7MVWS22KQTS4EMK7I46"
+    string to,
+    boolean toIsAccountID, // [required]
+
+    string amount, // Amount of the asset OR outside currency being sent. Example: "3.005"
+    // Zero or one of `asset` and `currency` should be set.
+    // If both are null or `asset` is native then XLM are being sent.
+    // If `currency` is set then XLM is being sent after converting `amount`.
+    // Note: Sending non-XLM assets is not yet supported.
+    union { null, OutsideCurrencyCode } currency,
+    union { null, Asset } asset,
+
+    string secretNote   // Encrypted for the sender, and recipient if possible.
```

Comment:
for requests secretNote is always visible to the recipient. Ends up in chat message.